### PR TITLE
UCS Server UUID pool modules and integration tests

### PR DESCRIPTION
**Pull request by dsoper2**
##### SUMMARY

Module for server UUID pool configuration on UCS Manager.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ucs_uuid_pool
##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (uuid_pool e929f71510) last updated 2018/06/20 11:44:41 (GMT -500)
  config file = /Users/dsoper/.ansible.cfg
  configured module search path = ['/Users/dsoper/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dsoper/Documents/ansible2/lib/ansible
  executable location = /Users/dsoper/Documents/ansible2/bin/ansible
  python version = 3.6.5 (default, Apr 20 2018, 18:22:17) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: ucs_uuid_pool
+short_description: Configures server UUID pools on Cisco UCS Manager
+description:
+- Configures server UUID pools and UUID blocks on Cisco UCS Manager.
+- Examples can be used with the UCS Platform Emulator U(https://communities.cisco.com/ucspe).
```
Comment (dagwieers) [inline lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py]: We recently added L(name,url) so you can actually do here:
```yaml
- Examples can be used with the L(UCS Platform Emulator,https://communities.cisco.com/ucspe).
```
which generally makes documentation more pleasing.

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    - The name of the UUID pool.
+    - This name can be between 1 and 32 alphanumeric characters.
+    - "You cannot use spaces or any special characters other than - (hyphen), \"_\" (underscore), : (colon), and . (period)."
+    - You cannot change this name after the UUID pool is created.
+    required: yes
+  description:
+    description:
+    - "The user-defined description of the UUID pool."
+    - Enter up to 256 characters.
+    - "You can use any characters or spaces except the following:"
```
Comment (dagwieers) [inline lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py:39]: This line does not need quotes. (nitpick)

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    - "` (accent mark), \ (backslash), ^ (carat), \" (double quote), = (equal sign), > (greater than), < (less than), or ' (single quote)."
+    aliases: [ descr ]
+  prefix:
+    description:
+    - UUID prefix used for the range of server UUIDs.
+    - "If no value is provided, the system derived prefix will be used (equivalent to selecting 'derived' option in UI)."
```
Comment (dagwieers) [inline lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py:45]: This line does not need quotes. (nitpick)

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    - "If the user provides a value, the user provided prefix will be used (equivalent to selecting 'other' option in UI)."
```
Comment (dagwieers) [inline lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py:46]: This line does not need quotes. (nitpick)

Comment (dagwieers) [reply lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py:46]: Below there are more like this, if you'd care :-)

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+                    ucs.login_handle.add_mo(mo, True)
+                    ucs.login_handle.commit()
+                changed = True
+
+    except Exception as e:
+        err = True
+        ucs.result['msg'] = "setup error: %s " % str(e)
+
+    ucs.result['changed'] = changed
```
Comment (dagwieers) [inline lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py]: I wonder why you are doing this here, and not just use ucs.result['changed'] on assignment ?

**Review by dagwieers — APPROVED** [review]
LGTM, only a few cosmetic-related comments. Nothing blocking.

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py (hunk 2 above)
Comment (dsoper2) [reply lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py:39]: With the colon, I had to quote for ansible-doc to accept the line.

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py (hunk 5 above)
Comment (dsoper2) [reply lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py]: updated with ucs.result on assignment

**Review by dsoper2 — COMMENTED** [review]
shipit

**Review by movinalot — APPROVED** [review]
shipit

File: lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py:39]: Oh right, trailing colon.

**Pull request merged**